matplotcheck/raster.py

```diff
@@ -114,19 +112,65 @@ def assert_legend_accuracy_classified_image(
         Finally those two arrays of strings are compared. Passes if they match.
         """
         # Retrieve image array
-        im_data = []
-        if self.ax.get_images():
-            im = self.ax.get_images()[0]
-            im_data, im_cmap = im.get_array(), im.get_cmap()
+        im_data = self.get_plot_image()
+        im = self.ax.get_images()[0]
+        im_cmap = im.get_cmap()
+
         assert list(im_data), "No Image Displayed"
```

Comment:
```suggestion
        assert list(im_data), "There is no image in this plot"
```

Reply:
Maybe add a comment above that says - `if the plot is empty, return error`

Reply:
get_plot_image() actually tests that an image exists. so i'm not sure why we even need this test here again. i'm still doubtful that this is the best way to test things but i do think a try: get_plot_image() is enough to ensure that this works. `try:` can caption the assertion error and fail gracefully.

We definitely want to add a few tests for plots with no image in them.